cfme/generic_objects/definition/button_groups.py

```diff
@@ -174,14 +207,30 @@ def delete(self, cancel=False):
         """
         view = navigate_to(self, 'Details')
 
-        if not view.toolbar.configuration.item_enabled('Remove this Button Group from Inventory'):
+        if not view.configuration.item_enabled('Remove this Button Group from Inventory'):
             raise OptionNotAvailable(
                 "Remove this Button Group is not enabled, there are buttons assigned to this group")
         else:
             view.configuration.item_select(
-                'Remove this Button Group from Inventory', handle_alert=not cancel)
-        view = self.create_view(GenericObjectDefinitionAllView)
-        assert view.is_displayed
+                'Remove this Button Group from Inventory', handle_alert=not cancel
+            )
+        view = self.create_view(GenericObjectDefinitionAllView, wait=10)
+        view.flash.assert_no_error()
+
+    def update(self, updates):
+        """Update this button in UI.
+
+        Args:
+            updates: Provided by update() context manager.
+        """
+        view = navigate_to(self, "Edit")
+        changed = view.fill(updates)
```

Comment:
@digitronik, you reinvent the wheel here :).
please look at fill_with method()

Reply:
I was unknown about this method . its awesome... I need to changes on other places I will push PR in next sprint for this... @izapolsk++